src/reducers/update.js

```diff
@@ -21,13 +21,17 @@ export const start = (state, action) => {
   const updatedRecordTuple = Map([[uid, updatedRecord]])
 
   return state.withMutations(map => {
-    // Optimistically update the record
-    map.set('raw', state.get('raw').merge(updatedRecordTuple))
+    if (action.isAsync) {
+      // Add updated item to pendingRaw
+      map.setIn(['pendingRaw', uid], updatedRecord)
```

Comment:
We should add to `pendingRaw` using the tuple.
```
map.set('pendingRaw', state.get('pendingRaw').merge(updatedRecordTuple))
```
Merging the tuple this way preserves the original uid type (string or integer)